test/e2e/e2e_test.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2020 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	// keeps downloaded content which doesn't change between test runs
+	cacheDir          = "/tmp/intel-device-plugins-for-kubernetes-test-cache"
+	kindBinaryPath    = cacheDir + "/kind"
+	kindURL           = "https://github.com/kubernetes-sigs/kind/releases/download/v0.6.1/kind-linux-amd64"
+	kindHash          = "386ef80ef8e1baeeca1fb8a8200736ff631958f5aaf9115383abfba973c76f5b"
+	kubectlBinaryPath = cacheDir + "/kubectl"
+	kubectlURL        = "https://storage.googleapis.com/kubernetes-release/release/v1.17.0/bin/linux/amd64/kubectl"
+	kubectlHash       = "6e0aaaffe5507a44ec6b1b8a0fb585285813b78cc045f8804e70a6aac9d1cb4c"
+	clusterName       = "intel-plugins-testing"
+	kubeConfigPath    = cacheDir + "/kube.config"
+	k8sImage          = "kindest/node:v1.17.0"
+	webhookImagePath  = cacheDir + "/intel-fpga-admissionwebhook-devel.tgz"
+	samplePod         = `
+apiVersion: v1
+kind: Pod
+metadata:
+  name: sample-pod
+  labels:
+    app: sample
+spec:
+  containers:
+  - name: sample-container
+    image: busybox
+    command: ['sh', '-c', 'echo Hello Kubernetes! && sleep 3600']
+    resources:
+      requests:
+        fpga.intel.com/arria10.dcp1.0-nlb0: 1
+      limits:
+        fpga.intel.com/arria10.dcp1.0-nlb0: 1
+
+`
+)
+
+func setUp(t *testing.T) {
+	info, err := os.Stat(cacheDir)
+	if err != nil || info == nil || !info.IsDir() {
+		os.RemoveAll(cacheDir)
+		os.MkdirAll(cacheDir, 0755)
+		fmt.Println("Downloading and caching required executables...")
+
+		for _, tuple := range [][]string{{kubectlBinaryPath, kubectlURL, kubectlHash}, {kindBinaryPath, kindURL, kindHash}} {
+			if err := fetchExecutable(tuple[0], tuple[1]); err != nil {
+				t.Fatalf("unable to fetch executable: %+v", err)
+			}
+			if err := checkHash(tuple[0], tuple[2]); err != nil {
+				t.Fatalf("unable to verify downloaded blob: %+v", err)
+			}
+		}
+		fmt.Println("Done.")
+	}
+	os.RemoveAll(webhookImagePath)
+	runCommand(t, "podman", "save", "docker.io/intel/intel-fpga-admissionwebhook:devel", "-o", webhookImagePath)
```

Comment:
I'd propose to add "docker.io/intel/intel-fpga-admissionwebhook:devel" to constants.